logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -593,6 +583,71 @@ private UptimeMetric initOrGetUptimeMetric(final ThreadContext context,
         return retrievedMetric.toJava(UptimeMetric.class);
     }
 
+    private void initializePqFlowMetrics(final ThreadContext context, final RubySymbol[] flowNamespace, final Metric<Number> uptimeInPreciseSeconds) {
+        final IRubyObject queueContext = getSetting(context, QueueFactoryExt.QUEUE_TYPE_CONTEXT_NAME);
+        if (!queueContext.isNil() && queueContext.asJavaString().equals(QueueFactoryExt.PERSISTED_TYPE)) {
+
+            final RubySymbol[] queueNamespace = buildNamespace(QUEUE_KEY);
+            final RubySymbol[] queueCapacityNamespace = buildNamespace(QUEUE_KEY, CAPACITY_KEY);
+
+            final Supplier<NumberGauge> eventsGaugeMetricSupplier = () -> initOrGetNumberGaugeMetric(context, queueNamespace, EVENTS_KEY).orElse(null);
+            final FlowMetric growthEventsFlow = createFlowMetric(QUEUE_PERSISTED_GROWTH_EVENTS_KEY, eventsGaugeMetricSupplier, () -> uptimeInPreciseSeconds);
+            this.flowMetrics.add(growthEventsFlow);
+            storeMetric(context, flowNamespace, growthEventsFlow);
+
+            final Supplier<NumberGauge> queueSizeInBytesMetricSupplier = () -> initOrGetNumberGaugeMetric(context, queueCapacityNamespace, QUEUE_SIZE_IN_BYTES_KEY).orElse(null);
+            final FlowMetric growthBytesFlow = createFlowMetric(QUEUE_PERSISTED_GROWTH_BYTES_KEY, queueSizeInBytesMetricSupplier, () -> uptimeInPreciseSeconds);
+            this.flowMetrics.add(growthBytesFlow);
+            storeMetric(context, flowNamespace, growthBytesFlow);
+        }
+    }
+
+    private void initializePluginFlowMetrics(final ThreadContext context, final Metric<Number> uptimeInPreciseMillis) {
+        for (IRubyObject rubyObject: lirExecution.inputs()) {
+            IRubyObject config = rubyObject.callMethod(context, "config");
+            String id = config.convertToHash().get("id").toString();
+            initializePluginThroughputFlowMetric(context, uptimeInPreciseMillis, id);
```

Comment:
Build failure source is here:

We have a `LogStash::JavaInputDelegator` (which is a java-`org.logstash.config.ir.compiler.JavaInputDelegatorExt`), which doesn;'t have a `config` method, but it _does_ have `@JRubyMethod(name = "id")` method.

~~~
04:23:55     [ERROR] 2022-11-16 12:23:55.605 [Converge PipelineAction::Create<main>] agent - Failed to execute action {:action=>LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"NoMethodError", :message=>"undefined method `config' for #<LogStash::JavaInputDelegator:0x7401e2a6>", :backtrace=>["org/logstash/execution/AbstractPipelineExt.java:529:in `initialize_flow_metrics'", "/opt/logstash/build/logstash-8.6.0-SNAPSHOT/logstash-core/lib/logstash/java_pipeline.rb:124:in `start'", "/opt/logstash/build/logstash-8.6.0-SNAPSHOT/logstash-core/lib/logstash/pipeline_action/create.rb:52:in `block in execute'", "/opt/logstash/build/logstash-8.6.0-SNAPSHOT/logstash-core/lib/logstash/pipelines_registry.rb:159:in `create_pipeline'", "/opt/logstash/build/logstash-8.6.0-SNAPSHOT/logstash-core/lib/logstash/pipeline_action/create.rb:51:in `execute'", "/opt/logstash/build/logstash-8.6.0-SNAPSHOT/logstash-core/lib/logstash/agent.rb:386:in `block in converge_state'"]}
04:23:55     [INFO ] 2022-11-16 12:23:55.783 [LogStash::Runner] runner - Logstash shut down.
04:23:55     Failed (
04:23:55     expected: 4
04:23:55          got: 0
04:23:55 
04:23:55     (compared using ==)
04:23:55     ). Sleeping for 0.02
04:24:27       successfully runs a pipeline with an installed Java plugins (FAILED - 1)
~~~

Since instances of classes extending ruby-`LogStash::Inputs::Base` _also_ have a ruby-`id` method, we can solve it with something like:

```suggestion
            IRubyObject rubyfiedId = rubyObject.callMethod(context, "id");
            String id = rubyfiedId.toString();
            initializePluginThroughputFlowMetric(context, uptimeInPreciseMillis, id);
```